src/components/expanded-state/token-history/TokenHistory.js

```diff
@@ -0,0 +1,329 @@
+import { useTheme } from '@rainbow-me/context/ThemeContext';
+import {
+  AccentColorProvider,
+  Box,
+  Inline,
+  Inset,
+  Stack,
+  Text,
+} from '@rainbow-me/design-system';
+import {
+  apiGetNftSemiFungibility,
+  apiGetNftTransactionHistoryForEventType,
+} from '@rainbow-me/handlers/opensea-api';
+import { web3Provider } from '@rainbow-me/handlers/web3';
+import { formatAssetForDisplay } from '@rainbow-me/helpers';
+import { getHumanReadableDate } from '@rainbow-me/helpers/transactions';
+import { useAccountProfile, useAccountSettings } from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import NetworkTypes from '@rainbow-me/networkTypes';
+import {
+  ENS_NFT_CONTRACT_ADDRESS,
+  REVERSE_RECORDS_MAINNET_ADDRESS,
+  reverseRecordsABI,
+} from '@rainbow-me/references';
+import Routes from '@rainbow-me/routes';
+import { handleSignificantDecimals } from '@rainbow-me/utilities';
+import { abbreviations } from '@rainbow-me/utils';
+import { EventTypes, PaymentTokens } from '@rainbow-me/utils/tokenHistoryUtils';
+
+const TokenHistory = ({ contractAddress, tokenID, accentColor }) => {
+  const [tokenHistory, setTokenHistory] = useState([]);
+  const { colors } = useTheme();
+  const { accountAddress } = useAccountProfile();
+  const { network } = useAccountSettings();
+  const { navigate } = useNavigation();
+
+  const networkPrefix = network === NetworkTypes.mainnet ? '' : `${network}-`;
+
+  useEffect(() => {
+    async function fetchTransactionHistory() {
+      const semiFungible = await apiGetNftSemiFungibility(
+        networkPrefix,
+        contractAddress,
+        tokenID
+      );
+
+      const [rawTransferEvents, rawSaleEvents] = await Promise.all([
+        apiGetNftTransactionHistoryForEventType(
+          networkPrefix,
+          semiFungible,
+          accountAddress,
+          contractAddress,
+          tokenID,
+          EventTypes.TRANSFER.type
+        ),
+        apiGetNftTransactionHistoryForEventType(
+          networkPrefix,
+          semiFungible,
+          accountAddress,
+          contractAddress,
+          tokenID,
+          EventTypes.SALE.type
+        ),
+      ]);
+
+      const rawEvents = rawTransferEvents.concat(rawSaleEvents);
+      const txHistory = await processRawEvents(contractAddress, rawEvents);
+
+      setTokenHistory(txHistory);
+    }
+    fetchTransactionHistory();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountAddress, contractAddress, networkPrefix, tokenID]);
+
+  const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  const reverseRecordContract = new Contract(
+    REVERSE_RECORDS_MAINNET_ADDRESS,
+    reverseRecordsABI,
+    web3Provider
+  );
+
+  const processRawEvents = async (contractAddress, rawEvents) => {
+    rawEvents.sort((event1, event2) =>
+      event2.created_date.localeCompare(event1.created_date)
+    );
+    let addressArray = [];
+    let sales = [];
```

Comment:
```suggestion
    const addressArray = [];
    const sales = [];
```

You're mutating the instance later, not changing the instance itself